postgres/src/config.rs

```diff
@@ -88,9 +89,9 @@ use tokio_postgres::{Error, Socket};
 /// ```not_rust
 /// postgresql:///mydb?user=user&host=/var/lib/postgresql
 /// ```
-#[derive(Clone)]
 pub struct Config {
     config: tokio_postgres::Config,
+    notification_callback: Box<dyn NotificationCallback>,
```

Comment:
This should just be able to store an `Arc<dyn FnMut(AsyncMessage) + Send>`.

Reply:
I tried that. Sadly an `Arc<dyn FnMut(AsyncMessage) + Send>` does not itself implement `Send`, as `Arc` propagates both `Send + Sync` iff its contents are `Send + Sync`. And without the `Send` bound you can't call the callback from a futures context.

I don't see a good way to do this besides this `Box<dyn NotificationCallback>` or the separate `connect_with` method I proposed originally.

My only requirement is to be able to easily use an MPSC transmitter from the notification callback. I don't have particular opinions on how the API should look, so if it'd be easier for you to just hammer on this PR to get into a mergeable shape, please feel free to do so!

Reply:
Ah right  yeah.